core/liquidity/v2/snapshot.go

```diff
@@ -0,0 +1,461 @@
+	return e.hashKeys
+}
+
+func (e *SnapshotEngine) GetState(k string) ([]byte, []types.StateProvider, error) {
+	state, err := e.serialise(k)
+	return state, nil, err
+}
+
+func (e *SnapshotEngine) LoadState(ctx context.Context, p *types.Payload) ([]types.StateProvider, error) {
+	if e.Namespace() != p.Data.Namespace() {
+		return nil, types.ErrInvalidSnapshotNamespace
+	}
+
+	switch pl := p.Data.(type) {
+	case *types.PayloadLiquidityV2Provisions:
+		return nil, e.loadProvisions(ctx, pl.Provisions.GetLiquidityProvisions(), p)
+	case *types.PayloadLiquidityV2Performances:
+		return nil, e.loadPerformances(pl.Performances, p)
+	case *types.PayloadLiquidityV2Supplied:
+		return nil, e.loadSupplied(pl.Supplied, p)
+	case *types.PayloadLiquidityV2Scores:
+		return nil, e.loadScores(pl.Scores, p)
+	default:
+		return nil, types.ErrUnknownSnapshotType
+	}
+}
+
+func (e *SnapshotEngine) Stopped() bool {
+	return e.stopped
+}
+
+func (e *SnapshotEngine) StopSnapshots() {
+	e.log.Debug("market has been cleared, stopping snapshot production", logging.MarketID(e.marketID))
+	e.stopped = true
+}
+
+func (e *SnapshotEngine) OnMinProbabilityOfTradingLPOrdersUpdate(d num.Decimal) {
+	e.Engine.OnMinProbabilityOfTradingLPOrdersUpdate(d)
+}
+
+func (e *SnapshotEngine) OnProbabilityOfTradingTauScalingUpdate(d num.Decimal) {
+	e.Engine.OnProbabilityOfTradingTauScalingUpdate(d)
+}
+
+func (e *SnapshotEngine) OnMaximumLiquidityFeeFactorLevelUpdate(d num.Decimal) {
+	e.Engine.OnMaximumLiquidityFeeFactorLevelUpdate(d)
+}
+
+func (e *SnapshotEngine) OnStakeToCcyVolumeUpdate(d num.Decimal) {
+	e.Engine.OnStakeToCcyVolumeUpdate(d)
+}
+
+func (e *SnapshotEngine) OnPriceRangeUpdate(d num.Decimal) {
+	e.Engine.OnPriceRangeUpdate(d)
+}
+
+func (e *SnapshotEngine) OnCommitmentMinTimeFractionUpdate(d num.Decimal) {
+	e.Engine.OnCommitmentMinTimeFractionUpdate(d)
+}
+
+func (e *SnapshotEngine) OnSlaCompetitionFactorUpdate(d num.Decimal) {
+	e.Engine.OnSlaCompetitionFactorUpdate(d)
+}
+
+func (e *SnapshotEngine) OnNonPerformanceBondPenaltySlopeUpdate(d num.Decimal) {
+	e.Engine.OnNonPerformanceBondPenaltySlopeUpdate(d)
+}
+
+func (e *SnapshotEngine) OnNonPerformanceBondPenaltyMaxUpdate(d num.Decimal) {
+	e.Engine.OnNonPerformanceBondPenaltyMaxUpdate(d)
+}
+
+func (e *SnapshotEngine) OnPerformanceHysteresisEpochsUpdate(d uint64) {
+	e.Engine.OnPerformanceHysteresisEpochsUpdate(d)
+}
+
+func (e *SnapshotEngine) serialise(k string) ([]byte, error) {
+	var (
+		buf     []byte
+		changed bool
+		err     error
+	)
+
+	switch k {
+	case e.provisionsKey:
+		buf, changed, err = e.serialiseProvisions()
+	case e.performancesKey:
+		buf, changed, err = e.serialisePerformances()
+	case e.suppliedKey:
+		buf, changed, err = e.serialiseSupplied()
+	case e.scoresKey:
+		buf, changed, err = e.serialiseScores()
+	default:
+		return nil, types.ErrSnapshotKeyDoesNotExist
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	if e.stopped {
+		return nil, nil
+	}
+
+	if !changed {
```

Comment:
can we not do this please - changed is by construction always true so can we remove it from the serialiseXXX, the marshal, and basically everywhere? I wonder if the linter isn't gonna tell you this before me...

Reply:
The linter didn't catch it, but still, you are right. Nice catch.